dataactbroker/handlers/accountHandler.py

```diff
@@ -137,6 +144,117 @@ def login(self,session):
+            max_xml = requests.get(url).content
+            max_dict = xmltodict.parse(max_xml)
+
+            if not 'cas:authenticationSuccess' in max_dict['cas:serviceResponse']:
+                raise ValueError("You have failed to login successfully with MAX")
+
+            # Grab the email and list of groups from MAX's response
+            email = max_dict['cas:serviceResponse']['cas:authenticationSuccess']['cas:attributes']['maxAttribute:Email-Address']
+            group_list_all = max_dict['cas:serviceResponse']['cas:authenticationSuccess']['cas:attributes']['maxAttribute:GroupList'].split(',')
+            group_list = list(filter(lambda g:  g.startswith(parent_group), group_list_all))
+
+            # Deny access if not in the parent group aka they're not allowed to access the website all together
+            if not parent_group in group_list:
+                raise ValueError("You have logged in with MAX but do not have permission to access the broker. Please "
+                                 "contact DATABroker@fiscal.treasury.gov to obtain access.")
+
+            cgac_group = list(filter(lambda g: len(g) == len(parent_group + "-CGAC_")+3, group_list))
+
+            # Deny access if they are not aligned with an agency
+            if not cgac_group:
+                raise ValueError("You have logged in with MAX but do not have permission to access the broker. Please "
+                                 "contact DATABroker@fiscal.treasury.gov to obtain access.")
+
+            try:
+                sess = GlobalDB.db().session
+                user = sess.query(User).filter(func.lower(User.email) == func.lower(email)).one_or_none()
+
+                # If the user does not exist, create them since they are allowed to access the site because they got
+                # past the above group membership checks
+                if user is None:
+                    user = User()
+
+                    first_name = max_dict["cas:serviceResponse"]['cas:authenticationSuccess']['cas:attributes'][
+                        'maxAttribute:First-Name']
+                    middle_name = max_dict["cas:serviceResponse"]['cas:authenticationSuccess']['cas:attributes'][
+                        'maxAttribute:Middle-Name']
+                    last_name = max_dict["cas:serviceResponse"]['cas:authenticationSuccess']['cas:attributes'][
+                        'maxAttribute:Last-Name']
+
+                    user.email = email
+
+                    # Check for None first so the condition can short-circuit without
+                    # having to worry about calling strip() on a None object
+                    if middle_name is None or middle_name.strip() == '':
+                        user.name = first_name + " " + last_name
+                    else:
+                        user.name = first_name + " " + middle_name[0] + ". " + last_name
+                    user.user_status_id = UserHandler().getUserStatusId('approved')
```

Comment:
This is a great reminder that I should communicate this more broadly--apologies! Part of the db refactor is using a new series of lookup dictionaries for our "helper tables". So for this, you can:
```python
from dataactcore.models.lookups import USER_STATUS_DICT
...
user.user_status_id = USER_STATUS_DICT['approved']
```

Essentially, we're now using a pre-defined set of key/value pairs to both load these database tables and to access the info, bypassing the db entirely (though we still load it for reference and to use for ad-hoc queries)